### ISSUE TITLE
[msh3 HTTP/3] Header without colon

### ISSUE THREAD

**Issue by gvanem** [curl/curl#8666]
Trying Nick's [MSH3](https://github.com/curl/curl/pull/8517) patches with `curl  --http3 https://www.google.com`, yields:
`curl: (8) Header without semicolon`

Turning on verbose mode, it seems to be a missing `;` in the `alt-svc` header `verify_header()` complains about:
```
< alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
```

174 bytes at line 11 (0-based). Adding this patch:

```diff
--- a/http.c 2022-04-01 14:25:11
+++ b/http.c 2022-04-01 15:14:23
@@ -3792,7 +3792,7 @@
   ptr = memchr(header, ':', hlen);
   if(!ptr) {
     /* this is bad, bail out */
-    failf(data, "Header without semicolon");
+    failf(data, "Header without semicolon at line %u", k->headerline);
     return CURLE_WEIRD_SERVER_REPLY;
   }
   return CURLE_OK;
```
I see it happens at line 15. But there are only 14 lines AFAICS (?):
  * line 12: `accept-ranges: none`
 * line 13: `vary: Accept-Encoding`  the last.

I know next to nothing about Alt-svc. or how `k->headerline` is used.

**Comment by bagder** [comment]
Hm, the check is for a colon - the message is wrong. All headers need a colon.

**Comment by gvanem** [comment]
Ok so the wording is wrong. But the `alt-svc:` has a colon. If not this, which one is `verify_header()` complaing about?

**Comment by bagder** [comment]
That's my question as well! :grin:

**Comment by bagder** [comment]
This is fixed now in #8517 with @nibanks's latest updates.
